src/axom/inlet/Field.cpp

```diff
@@ -56,14 +58,189 @@ bool Field::required()
+    SLIC_WARNING("Field value type did not match STRING");
+    setWarningFlag(m_sidreRootGroup);
+  }
+  setDefaultValue(value);
+  return shared_from_this();
+}
+
+std::shared_ptr<Field> Field::defaultValue(bool value) {
+  if (m_type != axom::sidre::DataTypeId::INT8_ID) {
+    SLIC_WARNING("Field value type did not match BOOL");
+    setWarningFlag(m_sidreRootGroup);
+  }
+  setDefaultValue((int8)value);
+  return shared_from_this();
+}
+
+std::shared_ptr<Field> Field::defaultValue(int value) {
+  switch (m_type) {
+  case axom::sidre::DataTypeId::INT_ID:
+    setDefaultValue(value);
+    break;
+  case axom::sidre::DataTypeId::DOUBLE_ID:
+    setDefaultValue((double)value);
+    break;
+  default:
+    SLIC_WARNING("Field value type did not match INT/DOUBLE");
+    setWarningFlag(m_sidreRootGroup);
+    break;
+  }
+  return shared_from_this();
+}
+
+std::shared_ptr<Field> Field::defaultValue(double value) {
+  if (m_type != axom::sidre::DataTypeId::DOUBLE_ID) {
+    SLIC_WARNING("Field value type did not match DOUBLE");
+    setWarningFlag(m_sidreRootGroup);
+  }
+  setDefaultValue(value);
+  return shared_from_this();
+}
+
+template<typename T>
+void Field::setRange(T startVal, T endVal) {
+  if (m_sidreGroup->hasView("range") || m_sidreGroup->hasView("validValues")
+      || m_sidreGroup->hasView("validStringValues")) {
+    std::string msg = fmt::format("Inlet Field has already defined range: {0}",
+                                  m_sidreGroup->getPathName());
+    SLIC_WARNING(msg);
+    setWarningFlag(m_sidreRootGroup);
+  } else {
+      auto* view = m_sidreGroup->createViewAndAllocate("range", m_type, 2);
+      T* pair = view->getArray();
+      pair[0] = startVal;
+      pair[1] = endVal;
+  }
+}
+
+std::shared_ptr<Field> Field::range(int startVal, int endVal) {
+  switch(m_type) {
+  case axom::sidre::DataTypeId::INT_ID:
+    this->setRange(startVal, endVal);
+    break;
+  case axom::sidre::DataTypeId::DOUBLE_ID:
+    this->setRange(static_cast<double>(startVal), static_cast<double>(endVal));  
+    break;
+  default: // incompatible type
+    SLIC_WARNING("Field value type did not match INT or DOUBLE");
+    setWarningFlag(m_sidreRootGroup);
+    break;
+  }
+  return shared_from_this();
+}
+
+std::shared_ptr<Field> Field::range(double startVal, double endVal) {
+  switch(m_type) {
+  case axom::sidre::DataTypeId::DOUBLE_ID:
+    this->setRange(startVal, static_cast<double>(endVal));  
+    break;
+  default: // incompatible type
+    SLIC_WARNING("Field value type did not match INT or DOUBLE");
+    setWarningFlag(m_sidreRootGroup);
+    break;
+  }
+  return shared_from_this();
+}
+
+std::shared_ptr<Field> Field::validValues(std::vector<int> set) {
+  if (m_type != axom::sidre::DataTypeId::INT_ID) {
+    SLIC_WARNING("Field value type did not match INT");
+    setWarningFlag(m_sidreRootGroup);
+  }
+
+  if (m_sidreGroup->hasView("range") || m_sidreGroup->hasView("validValues")
```

Comment:
Fix this message like previous code snippet (but switch the warnings to match this case) to clarify that they are setting valid values when range was set or that valid values was already set.  It's okay to bunch validValues and validStringValues warning messages together generically as "valid values" in the warning message.